VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts

```diff
@@ -34,20 +37,44 @@ describe('SelectHearingLayoutComponent', () => {
         videoCallService.getPreferredLayout.and.returnValue(layout);
         component.ngOnInit();
         expect(component.selectedLayout).toBe(layout);
+        expect(component.currentButtonContentKey).toBe(buttonContentKeyWhenOpen);
+    });
+
+    it('should show translated text on open accordion button', () => {
+        const expectedTranslatedContentForButton = 'this is translated for open all button';
+        component.currentButtonContentKey = buttonContentKeyWhenOpen;
+        translateServiceSpy.instant
+            .withArgs(`select-hearing-layout.${component.currentButtonContentKey}`)
+            .and.returnValue(expectedTranslatedContentForButton);
+        component.ngOnInit();
+        onLangChangeSpy.emit({ lang: 'tl' } as LangChangeEvent);
+        expect(component.accordionOpenAllElement.innerHTML).toContain(expectedTranslatedContentForButton);
+    });
+
+    it('should show translated text on open/close toggle button', () => {
+        const expectedTranslatedContentForHeader = 'this is translated for the accordion header';
+        translateServiceSpy.instant
+            .withArgs('select-hearing-layout.choose-hearing-layout')
+            .and.returnValue(expectedTranslatedContentForHeader);
+        component.ngOnInit();
+        onLangChangeSpy.emit({ lang: 'tl' } as LangChangeEvent);
+        expect(headingButton.innerHTML).toContain(expectedTranslatedContentForHeader);
     });
 
     it('should translate button text on text click', () => {
         component.ngOnInit();
         textButton.innerHTML = 'Close all';
```

Comment:
I don't think this line effects the test in any way does it?

Reply:
you are right :) removed the emit function